workflow-core/src/main/java/com/squareup/workflow/WorkflowIdentifier.kt

```diff
@@ -115,6 +131,25 @@ class WorkflowIdentifier internal constructor(
     return result
   }
 
+  /**
+   * Returns true if and only if both this identifier and [actual] have leaf real identifiers that
+   * have the following relationship:
+   *
+   *  - where the "leaf real identifier" is the last [ImpostorWorkflow.realIdentifier] in the chain
+   *    of [ImpostorWorkflow] identifiers that does not identify an [ImpostorWorkflow] itself,
+   *  - [actual]'s leaf identifier is the same type as, or a subtype of, this identifier's leaf
+   *    identifier.
+   *
+   * This predicate can be used for unit tests to assert that a workflow rendered a particular
+   * child workflow, given a supertype of that child workflow.
+   */
+  @TestOnly
+  fun matchesActualIdentifierForTest(actual: WorkflowIdentifier): Boolean {
```

Comment:
Is this not redundant with the public `val Workflow.identifier`, now that you've changed that to be based on `realIdentifier`? I guess the `isAssignableFrom` check is different than what the runtime normally needs to do?

Reply:
It's completely different. The regular runtime check does an exact match - the types must be exactly the same and the entire identifier chain must be equal recursively. This only looks at the last identifier and is an instanceOf vs ==.